Defined notes and rules for control BSI APP4.4.A12

benruland opened the pull request:
#### Description:

Notes / Rules for BSI APP4.4.A12 added.

This PR needs to be merged after #11342, as it builds upon the created skeleton in that PR.


#### Rationale:

As we have multiple customers asking for a BSI profile to be included in the compliance-operator, we are contributing a profile. To provide a better review process, the individual controle are implemented as separate PRs.

sluetze commented on controls/bsi_app_4_4.yml:
```diff
@@ -217,9 +217,12 @@ controls:
-    status: pending
-    rules: []
+      This requirement needs to be adressed in the respective separate systems.
+      However, one requirement can be checked automated: Encrypted communication all image registries.
```
Suggestion for line 221, since the grammar is a little bit off (bold from me to say that :D) :
`However, one requirement (Encrypted communication on all network ports) can partitially be checked by ensuring that no registry is allowed in over insecure protocols`

benruland replied:
Thanks for rewording, I took your version.

sluetze commented on controls/bsi_app_4_4.yml:
```diff
@@ -217,9 +217,12 @@ controls:
-    rules: []
+      This requirement needs to be adressed in the respective separate systems.
+      However, one requirement can be checked automated: Encrypted communication all image registries.
+    status: automated
```
following https://github.com/ComplianceAsCode/content/blob/master/docs/manual/developer/03_creating_content.md#reporting-status I would see the status as `partial`

benruland replied:
Indeed! Thanks for pointing out

sluetze reviewed (changes requested):
Hi @benruland,
please take a look at my comments and see if they fit.

jan-cerny commented:
The fail looks legit, you need to do what the test tells you to do, when you'll run that script it will update the product stability test data with the new reference URI you you just commit the change and it should get green.

benruland commented:
Thanks, I updated the test data as requested.

Surprisingly for me, the script also added new product_stability test data for openembedded, debian12 and anolis23. Hope this is expected.

yuumasato reviewed (approved):
LGTM

teacup-on-rockingchair reviewed (approved):
Great contribution :bow:

dodys reviewed (approved):
lgtm, thanks!

yuumasato commented:
At the moment there are no formal Oracle maintainers.
I'm overriding and merging.
